sentinelhub/constants.py

```diff
@@ -342,15 +342,22 @@ def from_string(mime_type_str):
         :return: A mime type enum
         :rtype: MimeType
         """
+        if '/' in mime_type_str:
```

Comment:
Before you do this, I think you can try calling `mimetypes.guess_extension`. This will handle all cases such as `text/plain`, `application/x-tar`, `application/x-hdf`, etc. Otherwise it seems weird that this is being handled in the dictionary below. If an input of this method is `plain` you probably don't want to parse it into `MimeType.TXT`, only if it is `text/plain`.

In case `mimetypes.guess_extension` returns `None`, then you try splitting by `'/'`. A case for that would be `application/tar`, which is for some applications a valid mime type but `mimetypes` doesn't acknowledge that.